### DISCUSSION TITLE
Questions on Windows support

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets#1624] [Q&A]
"It depends on µSockets, which is a standard C project for Linux, macOS & Windows."

it's a bit misleading to claim you support windows, if your internal library only comes with a Makefile, no obvious way to build in a Windows environment or with MSVC, and requires additional libraries (libuv) to even get it working on life support. this is getting into borderline "ofc it runs on windows, just install wsl" territory

**Comment by uNetworkingAB** [comment]
> vcpkg install uwebsockets

Above installs everything you need to start on Windows. We run all our CI tests on Windows, Linux, macOS with successful smoke test results.

You can see the latest results here (I think you have access to read the logs): https://github.com/uNetworking/uWebSockets/actions/runs/5103648567/jobs/9174036664

**Comment by ghost** [comment]
Thanks, I got it running - apologies for the previous rude comment.

It'd be nice to put a nod to this command somewhere in README - I imagine vcpkg is not the default way most people install a C++ library.

Also is there a reason not to include this file in the uSockets repo?
https://github.com/microsoft/vcpkg/blob/master/ports/usockets/CMakeLists.txt

**Comment by uNetworkingAB** [comment]
Well your criticism is not unmotivated. It's clear that we prefer Linux and macOS. But with vcpkg it becomes bearable on Windows. But Windows has never been a good C++ platform going back to the 90s. It always sucked in comparison. MSVC is the Internet Explorer of C++.
